zkvm/src/vm.rs

```diff
@@ -696,44 +696,25 @@ where
     }
 
     fn call(&mut self) -> Result<(), VMError> {
-        // Pop program, blinding factor, contract, and predicate
-        let prog = self.pop_item()?.to_data()?;
-        let blinding = self.pop_item()?.to_data()?;
+        // Pop program, call proof, and contract
+        let program_data = self.pop_item()?.to_data()?;
+        let program_item = program_data.clone().to_program_item()?;
```

Comment:
```suggestion
        let program_item = self.pop_item()?.to_data()?.to_program_item()?;
```

and let's remove unneeded `program_data `